kore/src/Kore/Builtin/Set.hs

```diff
@@ -335,9 +339,83 @@ evalDifference resultSort [_set1, _set2] = do
+                        commonSymbolicKeys
+                concreteKeys2 = Domain.getConcreteKeysOfAc _set2
+                commonConcreteKeys =
+                    filter (`Domain.isConcreteKeyOfAc` _set1) concreteKeys2
+                set1WithoutCommonKeys =
+                    foldr
+                        Domain.removeConcreteKeyOfAc
+                        set1withoutCommonSymbolicKeys
+                        commonConcreteKeys
+                set2WithoutCommonSymbolicKeys =
+                    foldr
+                        Domain.removeSymbolicKeyOfAc
+                        (Domain.getNormalizedSet _set2)
+                        commonSymbolicKeys
+                set2WithoutCommonKeys =
+                    foldr
+                        Domain.removeConcreteKeyOfAc
+                        set2WithoutCommonSymbolicKeys
+                        commonConcreteKeys
+                definedArgs =
+                    filter (not . TermLike.isDefinedPattern) args
+                    & map (makeCeilPredicate resultSort)
+                    & makeMultipleAndPredicate
+                    & Conditional.fromPredicate
+            patSet1 <-
+                Ac.returnAc
+                    resultSort
+                    (Domain.NormalizedSet set1WithoutCommonKeys)
+            patSet2 <-
+                Ac.returnAc
+                    resultSort
+                    (Domain.NormalizedSet set2WithoutCommonKeys)
+            let patDiff
+                  | set1WithoutCommonKeys == Domain.emptyNormalizedAc
+                    = patSet1
+                  | set2WithoutCommonKeys == Domain.emptyNormalizedAc
+                    = patSet1
+                  | otherwise = differenceSet <$> patSet1 <*> patSet2
+            return (flip Pattern.andCondition definedArgs patDiff)
+
+    rightIdentity <|> bothConcrete <|> symbolic
   where
     ctx = Set.differenceKey
+    binarySymbol :: Text -> Sort -> Internal.Symbol.Symbol
+    binarySymbol name symbolSort =
+        builtinSymbol name symbolSort [symbolSort, symbolSort]
+    builtinSymbol :: Text -> Sort -> [Sort] -> Internal.Symbol.Symbol
+    builtinSymbol name symbolSort operandSorts =
+        Internal.Symbol.Symbol
+            { symbolConstructor = TermLike.Id name TermLike.AstLocationTest
+            , symbolParams = []
+            , symbolAttributes = Default.def
+            , symbolSorts =
+                Internal.Symbol.applicationSorts operandSorts symbolSort
+            }
+        & Internal.Symbol.function
+    differenceSetSymbol :: Internal.Symbol.Symbol
+    differenceSetSymbol =
+        binarySymbol "differenceSet" resultSort
+        & Internal.Symbol.hook "SET.difference"
+    differenceSet
+        :: InternalVariable variable
+        => TermLike variable
+        -> TermLike variable
+        -> TermLike variable
+    differenceSet set1 set2 =
+        TermLike.mkApplySymbol differenceSetSymbol [set1, set2]
```

Comment:
You cannot copy these from the test suite. These are stand-ins for data supplied by the user, it cannot be hardcoded in the backend. You need the original symbol. `evalDifference` is eventually called by [`Kore.Builtin.Builtin.functionEvaluator`](https://github.com/kframework/kore/blob/292626439babae4b65bc4f9b3f6606fd3235c9d1/kore/src/Kore/Builtin/Builtin.hs#L271), which only passes the result sort. (That function call happens further down in this file.)

Instead, we should add a new function to `Kore.Builtin.Builtin`, something like this:

```.hs
applicationEvaluator :: _ -> BuiltinAndAxiomSimplifier
applicationEvaluator impl =
    applicationAxiomSimplifier evaluator
  where
    evaluator
        :: InternalVariable variable
        => MonadSimplify simplifier
        => Application Symbol (TermLike variable)
        -> simplifier (AttemptedAxiom variable)
    evaluator (_ :< app) = do
        let app' = fmap TermLike.removeEvaluated app
        getAttemptedAxiom (impl app' >>= appliedFunction)
```